workflows/dags/data/data_harvest.py

```diff
@@ -0,0 +1,221 @@
+                                f"{keyword}: {item[0]['lte']}-{item[0]['gte']}"
+                            )
+                    elif keyword == "observables":
+                        builder.add_keyword(f"{keyword}: {','.join(item)}")
+                    else:
+                        for value in item:
+                            builder.add_keyword(value)
+
+            builder = DataBuilder(source="hepdata")
+
+            builder.add_creation_date(datetime.datetime.now(datetime.UTC).isoformat())
+
+            base_record = payload["base"]
+
+            for collaboration in base_record["record"]["collaborations"]:
+                builder.add_collaboration(collaboration)
+
+            builder.add_abstract(base_record["record"]["data_abstract"])
+
+            add_keywords(base_record["record"], builder)
+
+            doi = base_record["record"].get("doi")
+
+            if doi:
+                builder.add_literature(
+                    doi={"value": doi},
+                    record={
+                        "$ref": f"https://inspirehep.net/api/literature/{base_record['record']['inspire_id']}"
+                    },
+                )
+            else:
+                builder.add_literature(
+                    record={
+                        "$ref": f"https://inspirehep.net/api/literature/{base_record['record']['inspire_id']}"
+                    },
+                )
+
+            for resource in base_record["record"]["resources"]:
+                if resource["url"].startswith(
+                    "https://www.hepdata.net/record/resource/"
+                ):
+                    continue
+                builder.add_url(resource["url"], description=resource["description"])
+
+            builder.add_title(base_record["record"]["title"])
+
+            builder.add_acquisition_source(
+                method="hepcrawl",
+                submission_number=base_record["record"]["inspire_id"],
+                datetime=datetime.datetime.now(datetime.UTC).isoformat(),
+            )
+
+            mtc = re.match(r"(.*?)\.v\d+", base_record["record"]["hepdata_doi"])
+            if mtc:
+                builder.add_doi(doi=mtc.group(1), material="data")
+            else:
+                builder.add_doi(
+                    doi=base_record["record"]["hepdata_doi"], material="data"
+                )
+
+            for _, record_version in payload.items():
+                add_version_specific_dois(record_version, builder)
+
+            data = builder.record
+            data["$schema"] = data_schema
+            return data
+
+        @task
+        def load_record(new_record):
+            """
+            Load the record to inspirehep.
+            """
+
+            try:
+                response = inspire_http_record_management_hook.get_record(
+                    pid_type="doi", control_number=new_record["dois"][0]["value"]
+                )
+            except RetryError:
```

Comment:
hmm `RetryError`?

Reply:
its to detect if the request fails, this is the generic error it throws when the doi does not exist. Will check if `raise_for_status` works in its place